syntax = "proto3";

package pa1;

service Inventory {
  rpc ProcessOrder (OrderRequest) returns (OrderReply);
}

enum OrderType {
  ORDER_TYPE_UNSPECIFIED = 0;
  GROCERY_ORDER = 1;
  RESTOCK_ORDER = 2;
}

enum Category {
  CATEGORY_UNSPECIFIED = 0;
  BREAD = 1;
  DAIRY = 2;
  MEAT = 3;
  PRODUCE = 4;
  PARTY = 5;
}

message Item {
  string name = 1;
  int32 qty = 2;
  Category category = 3;
}

message OrderRequest {
  string request_id = 1;
  OrderType order_type = 2;
  repeated Item items = 3;
}

message OrderReply {
  bool ok = 1;
  string message = 2;
}
